Sources/SwiftMetricsBluemix/SwiftMetricsBluemix.swift

```diff
@@ -443,21 +442,34 @@ public class SwiftMetricsBluemix {
 
   // Update local config from autoscaling service
   private func updateConfiguration(response: Data) {
-    let jsonData = JSON(data: response)
-    Log.debug("[Auto-scaling Agent] attempting to update configuration with \(jsonData)")
-    if (jsonData == nil) {
-      isAgentEnabled = false
-      return
-    }
-    if (jsonData["metricsConfig"]["agent"] == nil) {
-      isAgentEnabled = false
-      return
-    } else {
-      isAgentEnabled = true
-      enabledMetrics=jsonData["metricsConfig"]["agent"].arrayValue.map({$0.stringValue})
+    do {
+        let json = try JSONDecoder().decode(ConfigResponse.self, from: response)
+        Log.debug("[Auto-scaling Agent] attempting to update configuration with \(json)")
+        if json.agent.isEmpty {
+            isAgentEnabled = false
+            return
+        } else {
+            isAgentEnabled = true
+            self.enabledMetrics = json.agent
+        }
+        self.reportInterval = json.reportInterval
+        Log.exit("[Auto-scaling Agent] Updated configuration - enabled metrics: \(enabledMetrics), report interval: \(reportInterval) seconds")
+    } catch {
+        isAgentEnabled = false
+        return
     }
-    reportInterval=jsonData["reportInterval"].intValue
-    Log.exit("[Auto-scaling Agent] Updated configuration - enabled metrics: \(enabledMetrics), report interval: \(reportInterval) seconds")
   }
+    
+}
 
+public struct ConfigResponse: Codable {
+    var agent: [String]
+    var metricsConfig: String
+    var reportInterval: Int
+    
+    public enum CodingKeys: String, CodingKey {
```

Comment:
As these CodingKeys don't change the name of any of the keys, can we remove the enum and let the compiler synthesise it?